modules/snooker/src/main/ts/ephox/snooker/api/TableOperations.ts

```diff
@@ -149,19 +149,35 @@ const opMakeRowHeader = function (grid: Structs.RowCells[], detail: Structs.Deta
   return bundle(newGrid, detail.row, detail.column);
 };
 
-const opMakeColumnHeader = function (grid: Structs.RowCells[], detail: Structs.DetailExt, comparator: CompElm, genWrappers: GeneratorsTransform) {
-  const newGrid = TransformOperations.replaceColumn(grid, detail.column, comparator, genWrappers.replaceOrInit);
-  return bundle(newGrid, detail.row, detail.column);
+const opMakeColumnHeader = function (initialGrid: Structs.RowCells[], details: Structs.DetailExt[], comparator: CompElm, genWrappers: GeneratorsTransform) {
+  const columns = uniqueColumns(details);
+  let currentGrid = initialGrid;
+
+  Arr.each(columns, (column) => {
+    currentGrid = TransformOperations.replaceColumn(currentGrid, column.column, comparator, genWrappers.replaceOrInit);
+  });
+
+  const first = details[0];
+
+  return bundle(currentGrid, first ? first.row : 0, first ? first.column : 0);
```

Comment:
Strict null/undefined checks? as atm this is just a falsy check

Reply:
I thought the truthy checks would be suitable. I can change them into `Type.isUndefined` or `Type.isObject` if that'll read better.

Reply:
Sorry, you're right it's a truthy check, not a falsy check. Anyways, truthy/falsy checks should only be done for boolean variables. So yes, this should either use an `Optional` or use an explicit `Type` check.

My bigger question reading more now though, is why do you even need that check? It should never be an empty array from what I can see, as `onCells` guarantees it'll always have at least one value. You'll see everywhere else it's used, there's no check like this. eg: from `opInsertRowsBefore`:
```
  const example = details[0].row;
  const targetIndex = details[0].row;
```

Reply:
It shouldn't, but I also felt that it'd be poorly done to assume that there was something in there, since the input is from outside of the module and/or plugin.